torch/onnx/symbolic_opset9.py

```diff
@@ -835,24 +835,40 @@ def _prepare_onnx_paddings(dim, pad):
     paddings = paddings[-2::-2] + paddings[-1::-2]
     return paddings
 
+def _convert_padding_node(padding):
+    padding = sym_help._maybe_get_const(padding, 'is')
```

Comment:
Sorry I should've asked for sym_help._get_const here.
This one throws an error if all list elements are not const.
Maybe something like:
```
 try:
        padding = sym_help._get_const(padding, 'is', 'constant_pad_nd')
    except:
        _unimplemented('constant_pad_nd', "<msg>")
```

Reply:
I think `sym_help._get_const` won't work in this case, because this is the condition that it checks:
```
_is_value(value) and value.node().kind() not in ('onnx::Constant', 'prim::Constant')
```
and in the padding case, it can also be `prim::ListConstruct`.

But I did use your suggestion below for parsing the `value` in `constant_pad_nd`.

---
In reply to: [443761593](https://github.com/pytorch/pytorch/pull/39651#discussion_r443761593) [](ancestors = 443761593)

Reply:
How about after you unpack the list? Maybe like:

    try:
        padding = [sym_help._get_const(v, 'is', 'constant_pad_nd') for v in input_list]
    except:
        _unimplemented('constant_pad_nd', "<msg>")

It checks for prim:Constant type as well, also use of _unimplemented is more consistent.
I'm trying to avoid custom arg parsing or error handling for symbolics.